jwst/barshadow/tests/test_barshadow.py

```diff
@@ -0,0 +1,235 @@
+import numpy as np
+import numpy.random as rn
+
+from jwst import datamodels
+from jwst.barshadow import bar_shadow as bar
+
+
+def test_create_shutter_elements():
+
+    d1x1 = rn.random_sample((1001, 101))
+    d1x3 = rn.random_sample((1001, 101))
+    barshadow_model = datamodels.BarshadowModel(data1x1=d1x1, data1x3=d1x3)
+    shutter_elements = bar.create_shutter_elements(barshadow_model)
+
+    assert np.allclose(shutter_elements['first'], d1x1[:501, :], atol=1.e-10)
+    assert np.allclose(shutter_elements['open_open'], d1x3[:501, :],
+                       atol=1.e-10)
+    assert np.allclose(shutter_elements['open_closed'], d1x1[500:, :],
+                       atol=1.e-10)
+    assert np.allclose(shutter_elements['closed_open'], d1x1[:501, :],
+                       atol=1.e-10)
+    assert np.allclose(shutter_elements['closed_closed'],
+                       0.01 * np.ones((501, 101), dtype=np.float64),
+                       atol=1.e-10)
+    assert np.allclose(shutter_elements['last'], d1x1[501:, :], atol=1.e-10)
+
+
+def test_create_first():
+
+    shadow1x1 = rn.random_sample((1001, 101))
+
+    shadow = bar.create_first(shadow1x1)
+
+    assert np.allclose(shadow, shadow1x1[:501, :], atol=1.e-10)
+
+
+def test_create_open_open():
+
+    shadow1x3 = rn.random_sample((1001, 101))
+
+    shadow = bar.create_open_open(shadow1x3)
+
+    assert np.allclose(shadow, shadow1x3[:501, :], atol=1.e-10)
+
+
+def test_create_open_closed():
+
+    shadow1x1 = rn.random_sample((1001, 101))
+
+    shadow = bar.create_open_closed(shadow1x1)
+
+    assert np.allclose(shadow, shadow1x1[500:, :], atol=1.e-10)
+
+
+def test_create_closed_open():
+
+    shadow1x1 = rn.random_sample((1001, 101))
+
+    shadow = bar.create_closed_open(shadow1x1)
+
+    assert np.allclose(shadow, shadow1x1[:501, :], atol=1.e-10)
+
+
+def test_create_closed_closed():
+
+    shadow = bar.create_closed_closed()
+
+    assert np.allclose(shadow, 0.01 * np.ones((501, 101), dtype=np.float64),
+                       atol=1.e-10)
+
+
+def test_create_last():
+
+    shadow1x1 = rn.random_sample((1001, 101))
+
+    shadow = bar.create_last(shadow1x1)
+
+    assert np.allclose(shadow, shadow1x1[501:, :], atol=1.e-10)
+
+
+def test_create_shadow():
+
+    d1x1 = rn.random_sample((1001, 101))
+    d1x3 = rn.random_sample((1001, 101))
+    barshadow_model = datamodels.BarshadowModel(data1x1=d1x1, data1x3=d1x3)
+    shutter_elements = bar.create_shutter_elements(barshadow_model)
+
+    shutter_status = "0001x1000"
```

Comment:
I don't think you can have a shutter status with 0s at the ends, this slit would just be "1x1".

Reply:
I appreciate now that this is not realistic, but is this actually a problem?  Using that string tests all of the "open_closed", etc., options.

Reply:
True, for the purposes of the unit tests this is perfectly fine.